### PR TITLE
Adblock regional filters selection

### PR DESCRIPTION
**Pull request by emerick**
Fix brave/brave-browser#1931

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [x] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

### Default Region

- Start with a clean profile
- Set locale to "fr"
  - *Windows*: This is done in `chrome://settings/languages`
  - *Linux*: Run brave via `LANGUAGE=fr_FR ./brave`
  - *Mac*: Use the Language & Region operating system settings as described in https://github.com/brave/brave-browser/issues/1238#issuecomment-426068531
- Restart Brave
- Visit `brave://adblock`
- Verify that French regional list is selected
- Test some French regional sites
- Uncheck the French regional list
- Restart Brave
- Visit `brave://adblock` and verify that the French regional list remains unchecked

### Regular Usage

- Visit `brave://adblock`
- Enable one or more regional lists by clicking the associated checkbox
- Visit `brave://components` and verify that the enabled lists show up as installed components
- Test some regional sites, here are the lists https://github.com/brave/ad-block/blob/master/lists/regions.h you can load a list URL (e.g., https://easylist-downloads.adblockplus.org/Liste_AR.txt) to see the resources which it blocks
- Disable one or more regional lists by unchecking the associated checkbox
- Restart Brave
- Visit `brave://components` and verify that the disabled lists no longer show up as installed components

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: browser/extensions/brave_component_extension.h [hunk 1]
```diff
@@ -26,6 +29,8 @@ class BraveComponentExtension : public ComponentsUI {
                 const std::string& component_base64_public_key);
   static bool Unregister(const std::string& component_id);
 
+  std::string component_id() const;
```
Comment (bbondy) [inline browser/extensions/brave_component_extension.h]: no objection about this but I don't think it's used?

File: components/brave_shields/browser/ad_block_regional_service_manager.cc [hunk 2]
```diff
@@ -49,11 +49,24 @@ void AdBlockRegionalServiceManager::StartRegionalServices() {
...
+  bool checked_default_region =
+      local_state->GetBoolean(kAdBlockCheckedDefaultRegion);
+  if (!checked_default_region) {
+    local_state->SetBoolean(kAdBlockCheckedDefaultRegion, true);
+    auto it = brave_shields::FindAdBlockFilterListByLocale(
+        region_lists, g_brave_browser_process->GetApplicationLocale());
+    if (it == region_lists.end())
+      return;
+    EnableFilterList(it->uuid, true);
+  }
```
Comment (bbondy) [inline components/brave_shields/browser/ad_block_regional_service_manager.cc]: good

File: components/brave_shields/browser/ad_block_regional_service_manager.cc [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    const std::string& tab_host,
+    bool* matching_exception_filter,
+    bool* cancel_request_explicitly) {
+  base::AutoLock lock(regional_services_lock_);
+  for (const auto& regional_service : regional_services_) {
+    if (!regional_service.second->ShouldStartRequest(
+            url, resource_type, tab_host, matching_exception_filter,
+            cancel_request_explicitly)) {
+      return false;
+    }
```
Comment (bbondy) [inline components/brave_shields/browser/ad_block_regional_service_manager.cc]: We need to `return true` here as soon as `matching_exception_filter` is filled out for any of the instances.

File: browser/extensions/brave_component_extension.h (hunk 1 above)
Comment (emerick) [reply browser/extensions/brave_component_extension.h]: Good catch, I'll remove that it turned out not to be necessary.

**Pull request merged**